src/python/WMCore/WMSpec/Steps/Executors/DQMUpload.py

```diff
@@ -293,3 +311,253 @@ def upload(self, url, args, filename):
+        """
+        Register a file to new DQM GUi
+        - https://github.com/cms-DQM/dqmgui
+        HTTP request body:
+            [{"dataset": "/a/b/c", "run": "123456", "lumi": "0", "file": "/eos/cms/store/group/comm_dqm/DQMGUI_data/location/file.root", "fileformat": 1}]
+        Set lumis to 0, as this is reproducing current per Run based root files
+        - https://github.com/dmwm/WMCore/issues/10287#issuecomment-1052558061
+        DQM root files produced by Harvesting processing are type 1 (plain ROOT)
+        - https://github.com/cms-DQM/dqmgui#file-formats
+
+        :param stepName: str, Name of the step we are running through
+        :param analysisFile: WMCore.Configuration.ConfigSection, Analysis file in this step from WMCore step report
+        :return:
+        """
+        # Get task description
+        jobBag = self.job.getBaggage()
+        datasetName = self.job.get('inputDataset', None)
+
+        # Get runNumber, depending on the harvesting job mode (byRun vs multiRun)
+        multiRun = getattr(jobBag, "multiRun", False)
+        forceRunNumber = getattr(jobBag, "forceRunNumber", 999999)
+        if multiRun:
+            runNumber = forceRunNumber
+        else:
+            try:
+                runNumber = list(self.job['mask']['runAndLumis'].keys())[0]
+            except Exception as ex:
+                msg = "Error while retrieving run number from job definition:\n %s" % str(ex)
+                raise WMExecutionFailure(70320, "DQMUploadFailure", msg)
+
+        args = {}
+        args['file'] = self.registerEOSPrefix + \
+                       self.getAnalysisFileLFN(analysisFile)
+        args['dataset'] = datasetName
+        args['run'] = runNumber
+        args['lumi'] = 0
+        args['fileformat'] = 1
+
+        msg = "HTTP Upload is about to start:\n"
+        msg += " => URL: %s\n" % self.registerURL
+        msg += " => Filename: %s\n" % args['file']
+        logging.info(msg)
+
+        for numRetry in range(self.retryCount + 1):
+            try:
+                logException = True
+                result = self._register(self.registerURL, [args])
+                msg = '  Status code: %s\n' % result.getcode()
+                if result.getcode() >= 400 or result.getcode() is None:
+                    logException = False
+                    raise Exception(msg)
+                else:
+                    msg = 'HTTP POST to register url finished succesfully with response:\n' + msg
+                    logging.info(msg)
+                    break
+            except Exception as ex:
+                if numRetry == self.retryCount:
+                    msg = 'HTTP POST to register url failed! Error:\n%s' % str(ex)
+                    if logException:
+                        logging.exception(msg)
+                    else:
+                        logging.error(msg)
+                    raise WMExecutionFailure(70319, "DQMUploadFailure", msg)
```

Comment:
Please add this new error code and message to the WMExceptions.py module as well.